docs/source/blueprints/BP002.rst

```diff
@@ -0,0 +1,220 @@
+standard library. Using it to represent requirements is flexible and
+straightforward.
+
+The standard proposed way to represent requirements with JSON is defining one
+requirement per entry. Each entry should start with the requirement type,
+followed by other keyword arguments related to that type. Example::
+
+        [
+            {"type": "url", "url": "https://cabort.com/cabort.c", "hash": "deadbeefdeadbeef"},
+            {"type": "vmimage", "distro": "fedora", "version": 31, "arch": "x86_64"},
+            {"type": "package", "package": "lvm"}
+        ]
+
+The requirement `type` should match the module responsible for that type of
+requirement.
+
+Requirements representation as Python executable
+------------------------------------------------
+
+Another way to create the requirements representation as JSON files is by
+writing a Python executable. This approach makes the requirements
+representation flexible, by allowing the use of Python variables and code that
+may change the parameters values for the requirements, depending on the
+environment the Python code runs.
+
+The following example shows a requirement that depends on the architecture
+the test is running::
+
+        #!/usr/bin/python3
+
+        import os
+        import json
+
+        requirements = [
+            {"type": "url", "url": "https://cabort.com/cabort.c", "hash": "deadbeefdeadbeef"},
+            {"type": "vmimage", "distro": "fedora", "version": 31, "arch": os.uname()[4]},
+            {"type": "package", "package": "lvm"}
+        ]
+
+        print(json.dumps(requirements))
+
+Requirements representation as metadata on test docstring
+---------------------------------------------------------
+
+Test writers may want to add the requirements of a test into de test code. The
+option proposed here allows the use of metadata on test docstrings to represent
+the requirements list.
+
+Below is an example of how to define requirements as metadata on docstrings::
+
+        def test_something(self):
+        '''
+            :avocado requirement: [{"type": "url", "url": "https://cabort.com/cabort.c", "hash": "deadbeefdeadbeef"}]
+            :avocado requirement: [{"type": "vmimage", "distro": "fedora", "version": 31, "arch": "x86_64"}]
+            :avocado requirement: [{"type": "package", "package": "lvm"}]
+        '''
+           <test code>
+
+Requirements files location
+===========================
+
+It may be useful for test writers to define a standard source location for the
+requirements JSON files and the requirements Python executable.
+
+This blueprint defines the default location for those files into a requirements
+directory preceded by the test file name. It is also supported the use of
+sub-directories using the name of a specific test to define requirements for
+that test.
+
+The following file tree is an example of possible use for requirements
+directories for a test::
+
+        cabort.py
+        cabort.py.requirements/
```

Comment:
Just a personal "taste" here:

Any reason to use this convention? instead of a simple `.requirements/` dir? (without the prefix)? And inside this "requirements dir" we define the requirements based on the file names? Ex:
```
.requirements/main.json
.requirements/cabort.json
.requirements/cabort.test_2.json
```
Not sure here, just an idea. Trying to keep not so "polluted".

Reply:
> Any reason to use this convention?

I followed the convention Avocado has for the `data` dir, like, for example, `cabort.py.data`. I don't mind changing it, with the condition that we also change the `data` convention to keep things standardized.

What I don't think user-friendly to the test writers is the use of hidden directories. When I see a hidden directory, I consider them as part of the application, but, in the case of Avocado tests, the test writer is responsible for the requirements directory. Making the requirements directory hidden may confuse test writers that just touched a written test with requirements under this directory.

Reply:
It was just an idea, not need to worry for now, unless that is an acceptable idea.

Reply:
I share @willianrampazzo feelings about hidden files.  I'm all in favor, though, of a different requirements directory layout.

On the other hand, the positive thing about following a model that has been tried and tested for test "data" files, makes me believe that it will probably behave well for "requirements" too.  And by following that model, we can focus on the things that have not been solved yet.

Reply:
I agree @clebergnu , we could have an open issue to discuss this later.

It is just an idea to avoid a polluted directory.